web-client/public/js/update-app.js

```diff
@@ -212,52 +232,65 @@ export const updateApp = grnState => {
-        grnState.linkDistanceSlider.backup = grnState.linkDistanceSlider.currentVal;
-        grnState.chargeSlider.currentVal = grnState.chargeSlider.defaultVal;
-        grnState.linkDistanceSlider.currentVal = grnState.linkDistanceSlider.defaultVal;
-        $("#charge-menu").val(grnState.chargeSlider.defaultVal);
-        $("#link-distance-menu").val(grnState.linkDistanceSlider.defaultVal);
+        CHARGE_BACKUP_VALUE = CHARGE_CURRENT_VALUE;
+        LINK_DIST_BACKUP_VALUE = LINK_DIST_CURRENT_VALUE;
+        CHARGE_CURRENT_VALUE = CHARGE_DEFAULT_VALUE;
+        LINK_DIST_CURRENT_VALUE = LINK_DIST_DEFAULT_VALUE;
+        $(CHARGE_MENU).val(CHARGE_DEFAULT_VALUE);
+        $(LINK_DIST_MENU).val(LINK_DIST_DEFAULT_VALUE);
     };
 
     const undoReset = () => {
-        grnState.chargeSlider.currentVal = grnState.chargeSlider.backup;
-        grnState.linkDistanceSlider.currentVal = grnState.linkDistanceSlider.backup;
-        $("#charge-menu").val(grnState.chargeSlider.backup);
-        $("#link-distance-menu").val(grnState.linkDistanceSlider.backup);
+        CHARGE_CURRENT_VALUE = CHARGE_BACKUP_VALUE;
+        LINK_DIST_CURRENT_VALUE = LINK_DIST_BACKUP_VALUE ;
+        $(CHARGE_MENU).val(CHARGE_BACKUP_VALUE);
+        $(LINK_DIST_MENU).val(LINK_DIST_BACKUP_VALUE );
+    };
+
+    const resetForce = () => {
+        modifyChargeParameter(CHARGE_DEFAULT_VALUE);
+        modifyLinkDistanceParameter(LINK_DIST_DEFAULT_VALUE);
+    };
+
+    const undoResetForce = () => {
+        modifyChargeParameter(CHARGE_BACKUP_VALUE);
+        modifyLinkDistanceParameter(LINK_DIST_BACKUP_VALUE);
     };
 
     const updateChargeSliderValues = () => {
-        $(grnState.chargeSlider.sliderId).val(grnState.chargeSlider.currentVal);
-        $(grnState.chargeSlider.valueId).html(grnState.chargeSlider.currentVal +
-          ((grnState.chargeSlider.needsAppendedZeros && grnState.chargeSlider.currentVal.toString().length ===
-          GRAVITY_LENGTH_WITHOUT_ZERO) ? "0" : ""));
+        $(CHARGE_SLIDER_ID).val(CHARGE_CURRENT_VALUE);
+        $(CHARGE_SLIDER_ID).html(CHARGE_CURRENT_VALUE +
+          ((grnState.chargeSlider.needsAppendedZeros
+              && CHARGE_CURRENT_VALUE.toString().length === GRAVITY_LENGTH_WITHOUT_ZERO) ? "0" : ""));
     };
 
     const updateLinkDistanceSliderValues = () => {
-        $(grnState.linkDistanceSlider.sliderId).val(grnState.linkDistanceSlider.currentVal);
-        $(grnState.linkDistanceSlider.valueId).html(grnState.linkDistanceSlider.currentVal +
+        $(LINK_DIST_SLIDER_ID).val(LINK_DIST_CURRENT_VALUE);
+        $(LINK_DIST_SLIDER_ID).html(LINK_DIST_CURRENT_VALUE +
           ((grnState.linkDistanceSlider.needsAppendedZeros
-            && grnState.linkDistanceSlider.currentVal.toString().length ===
-              GRAVITY_LENGTH_WITHOUT_ZERO) ? "0" : ""));
+            && LINK_DIST_CURRENT_VALUE.toString().length === GRAVITY_LENGTH_WITHOUT_ZERO) ? "0" : ""));
```

Comment:
These lines (275-277) follow the same pattern as lines 267-270, so that looks like a good candidate for "pulling out" as a reusable function.